src/compare.py

```diff
@@ -128,41 +128,33 @@ def compare_phash(source: MatLike, capture: MatLike, mask: MatLike | None = None
     return 1 - (hash_diff / 64.0)
 
 
-def extract_and_compare_text(capture: MatLike, texts: list[str], method_index: int):
+def extract_and_compare_text(capture: MatLike, texts: list[str], methods_index: list[int]):
```

Comment:
Non-blocking, but these could be `Iterable`

```suggestion
def extract_and_compare_text(capture: MatLike, texts: Iterable[str], methods_index: Iterable[int]):
```